### ISSUE TITLE
exception when setting up new env using py 3.11

### ISSUE THREAD

**Issue by meerkatters** [glyph/Pomodouroboros#55]
```
(py24_sprint) ➜  Pomodouroboros git:(pycon24_sprint) pip install -e .   
Obtaining file:///Users/meer/proj/Pomodouroboros
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build editable did not run successfully.
  │ exit code: 1
  ╰─> [56 lines of output]
      Traceback (most recent call last):
        File "/Users/meer/my_venv/py24_sprint/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Users/meer/my_venv/py24_sprint/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/meer/my_venv/py24_sprint/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 132, in get_requires_for_build_editable
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/k3/9s5ywy351flfgnc6c61ffm140000gn/T/pip-build-env-i6t_xbcs/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 448, in get_requires_for_build_editable
          return self.get_requires_for_build_wheel(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/k3/9s5ywy351flfgnc6c61ffm140000gn/T/pip-build-env-i6t_xbcs/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/k3/9s5ywy351flfgnc6c61ffm140000gn/T/pip-build-env-i6t_xbcs/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/private/var/folders/k3/9s5ywy351flfgnc6c61ffm140000gn/T/pip-build-env-i6t_xbcs/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 40, in <module>
        File "/private/var/folders/k3/9s5ywy351flfgnc6c61ffm140000gn/T/pip-build-env-i6t_xbcs/overlay/lib/python3.11/site-packages/setuptools/__init__.py", line 103, in setup
          return distutils.core.setup(**attrs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/k3/9s5ywy351flfgnc6c61ffm140000gn/T/pip-build-env-i6t_xbcs/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 146, in setup
          _setup_distribution = dist = klass(attrs)
                                       ^^^^^^^^^^^^
        File "/private/var/folders/k3/9s5ywy351flfgnc6c61ffm140000gn/T/pip-build-env-i6t_xbcs/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 307, in __init__
          _Distribution.__init__(self, dist_attrs)
        File "/private/var/folders/k3/9s5ywy351flfgnc6c61ffm140000gn/T/pip-build-env-i6t_xbcs/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 284, in __init__
          self.finalize_options()
        File "/private/var/folders/k3/9s5ywy351flfgnc6c61ffm140000gn/T/pip-build-env-i6t_xbcs/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 658, in finalize_options
          for ep in sorted(loaded, key=by_order):
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/k3/9s5ywy351flfgnc6c61ffm140000gn/T/pip-build-env-i6t_xbcs/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 657, in <lambda>
          loaded = map(lambda e: e.load(), filtered)
                                 ^^^^^^^^
        File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/importlib/metadata/__init__.py", line 202, in load
          module = import_module(match.group('module'))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/importlib/__init__.py", line 126, in import_module
          return _bootstrap._gcd_import(name[level:], package, level)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
        File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
        File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
        File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
        File "<frozen importlib._bootstrap_external>", line 940, in exec_module
        File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
        File "/private/var/folders/k3/9s5ywy351flfgnc6c61ffm140000gn/T/pip-build-env-i6t_xbcs/overlay/lib/python3.11/site-packages/py2app/build_app.py", line 37, in <module>
          from py2app import recipes
        File "/private/var/folders/k3/9s5ywy351flfgnc6c61ffm140000gn/T/pip-build-env-i6t_xbcs/overlay/lib/python3.11/site-packages/py2app/recipes/__init__.py", line 10, in <module>
          from . import matplotlib  # noqa: F401
          ^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/k3/9s5ywy351flfgnc6c61ffm140000gn/T/pip-build-env-i6t_xbcs/overlay/lib/python3.11/site-packages/py2app/recipes/matplotlib.py", line 5, in <module>
          from pkg_resources import packaging
      ImportError: cannot import name 'packaging' from 'pkg_resources' (/private/var/folders/k3/9s5ywy351flfgnc6c61ffm140000gn/T/pip-build-env-i6t_xbcs/overlay/lib/python3.11/site-packages/pkg_resources/__init__.py)
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build editable did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

**Comment by glyph** [comment]
This is caused by https://github.com/ronaldoussoren/py2app/issues/525

**Comment by glyph** [comment]
we need a documented process for setting up new people, although the proximal cause here was addressed by f4b7cffac69a9b58b45536e573fc46e53cfe1fa0

**Comment by glyph** [comment]
The upstream issue is fixed and we are no longer pinned, so I think this is fixed!
